File: articles/grit-setbacks-and-recovery/go/grit_setbacks_recovery_score.go
package main

import "fmt"

func GritScore(perseveranceEffort, consistencyInterests float64) float64 {
	return 0.60*perseveranceEffort + 0.40*consistencyInterests
}

func RecoveryCapacityScore(
	emotionalRecovery,
	cognitiveRecovery,
	physicalRestoration,
	socialSupport,
	practicalResources float64,
) float64 {
	return 0.22*emotionalRecovery +
		0.22*cognitiveRecovery +
		0.18*physicalRestoration +
		0.20*socialSupport +
		0.18*practicalResources
}

func AdaptivePersistenceIndex(
	grit,
	setbackSeverity,
	recoveryCapacity,
	feedbackQuality,
	opportunityAccess,
	burnout float64,
) float64 {
	return 0.18*grit -
		0.22*setbackSeverity +
		0.28*recoveryCapacity +
		0.18*feedbackQuality +
		0.18*opportunityAccess -
		0.20*burnout
}

func main() {
	grit := GritScore(0.82, 0.64)
	recovery := RecoveryCapacityScore(0.74, 0.82, 0.66, 0.58, 0.62)
	persistence := AdaptivePersistenceIndex(grit, -0.28, recovery, 0.61, 0.70, -0.30)

	fmt.Printf("Synthetic grit score: %.3f\n", grit)
	fmt.Printf("Synthetic recovery capacity score: %.3f\n", recovery)
	fmt.Printf("Synthetic adaptive persistence index: %.3f\n", persistence)
}

File: articles/grit-setbacks-and-recovery/fortran/grit_setbacks_recovery_score.f90
program grit_setbacks_recovery_score
  implicit none
  real :: perseverance_effort, consistency_interests, grit
  real :: emotional_recovery, cognitive_recovery, physical_restoration, social_support, practical_resources
  real :: setback_severity, feedback_quality, opportunity_access, burnout
  real :: recovery, persistence

  perseverance_effort = 0.82
  consistency_interests = 0.64
  emotional_recovery = 0.74
  cognitive_recovery = 0.82
  physical_restoration = 0.66
  social_support = 0.58
  practical_resources = 0.62
  setback_severity = -0.28
  feedback_quality = 0.61
  opportunity_access = 0.70
  burnout = -0.30

  grit = 0.60 * perseverance_effort + 0.40 * consistency_interests
  recovery = 0.22 * emotional_recovery + 0.22 * cognitive_recovery + &
             0.18 * physical_restoration + 0.20 * social_support + &
             0.18 * practical_resources

  persistence = 0.18 * grit - 0.22 * setback_severity + 0.28 * recovery + &
                0.18 * feedback_quality + 0.18 * opportunity_access - 0.20 * burnout

  print *, "Synthetic grit score:", grit
  print *, "Synthetic recovery capacity score:", recovery
  print *, "Synthetic adaptive persistence index:", persistence
end program grit_setbacks_recovery_score
